polaris-react/src/components/Modal/components/Header/Header.tsx

```diff
@@ -13,16 +16,41 @@ export interface HeaderProps {
 }
 
 export function Header({id, titleHidden, children, onClose}: HeaderProps) {
+  const titleHiddenMarkup = (
+    <div className={styles.titleHidden}>
+      <Inline align="end">
+        <CloseButton titleHidden={titleHidden} onClick={onClose} />
+      </Inline>
+    </div>
+  );
+
+  if (titleHidden || !children) {
+    return titleHiddenMarkup;
+  }
+
   return (
-    <div
-      className={titleHidden || !children ? styles.titleHidden : styles.Header}
+    <Box
+      paddingTop="4"
+      paddingBottom="4"
+      paddingLeft="5"
+      paddingRight="5"
+      borderBottom="divider"
     >
-      <div id={id} className={styles.Title}>
-        <DisplayText element="h2" size="small">
-          {children}
-        </DisplayText>
-      </div>
-      <CloseButton titleHidden={titleHidden} onClick={onClose} />
-    </div>
+      <Columns columns={{xs: '1fr auto'}}>
+        <Inline>
+          <Box id={id} paddingTop="1">
+            {/* Replace with <Text> once responsive styles supported */}
+            <div className={styles.Title}>
+              <DisplayText element="h2" size="small">
```

Comment:
`Text` has a `breakWord` prop so it could replace `DisplayText` here? And the `styles.Title` are no longer needed?

Reply:
I originally used `Text` but it didn't have responsive styling that `DisplayText` has, which is why I left the `div` and `DisplayText`.
I just shipped the PR to add responsive styling to `Text` so I'll rebase and update this.